app/components/markdown/markdown_image/index.tsx

```diff
@@ -50,42 +56,63 @@ const style = StyleSheet.create({
 
 const MarkdownImage = ({
     disabled, errorTextStyle, imagesMetadata, isReplyPost = false,
-    linkDestination, postId, source,
+    linkDestination, location, postId, source,
 }: MarkdownImageProps) => {
     const intl = useIntl();
     const isTablet = useIsTablet();
     const theme = useTheme();
     const managedConfig = useManagedConfig();
-    const genericFileId = useRef(generateId()).current;
+    const genericFileId = useRef(generateId('uid')).current;
```

Comment:
error prone cause? can you elaborate here?

Reply:
Refs are expected to be usable anywhere regardless of the potential changes in the value. So people seeing `genericFieldId` being apparently a ref, may use it without realizing it is not a ref, but the ref value. Therefore, passing this to a callback will always send the value of the current render, not the reference to access the value in that particular moment.

In this particular case, the value will never change between renders, so it is the same, but this pattern may lead to errors when someone else takes over this code.

Reply:
that is totally expected.. not every ref is about a component, specially true with animations, the concept of refs is normally associated with a component but it is definitely not the case.. I would have it as a variable but it could reset it's value which is something we want to avoid.